packages/core/src/StateMachine.ts

```diff
@@ -450,6 +450,17 @@ export class StateMachine<
       }
     });
 
+    const context = JSON.parse(JSON.stringify(state.context), (_, value) => {
```

Comment:
This should be implemented as a custom traversal thing (that clones the affected object paths). Piggy-backing like this on `JSON.parse`+`JSON.stringify` like this is too simple of a solution that won't cover for a lot of situations. 

The persisted state should be serializable - but that doesn't necessarily mean that it has to be JSON-serializable. What if users use a different format (like even `superjson`)?